go/vt/vtorc/logic/topology_recovery.go

```diff
@@ -663,7 +663,12 @@ func executeCheckAndRecoverFunction(analysisEntry inst.ReplicationAnalysis, cand
 		// of a shard because a new tablet could have been promoted, and we need to have this visibility before we
 		// run a cluster operation of our own.
 		if isClusterWideRecovery(checkAndRecoverFunctionCode) {
-			forceRefreshAllTabletsInShard(ctx, analysisEntry.AnalyzedKeyspace, analysisEntry.AnalyzedShard)
+			log.Infof("isClusterWideRecovery ---> 1, %+v detection", analysisEntry.Analysis)
+			if analysisEntry.Analysis == inst.DeadPrimary {
+				forceRefreshAllTabletsInShard(ctx, analysisEntry.AnalyzedKeyspace, analysisEntry.AnalyzedShard, true)
+			} else {
+				forceRefreshAllTabletsInShard(ctx, analysisEntry.AnalyzedKeyspace, analysisEntry.AnalyzedShard, false)
```

Comment:
We should check the `checkAndRecoverFunctionCode` instead of the `analysisEntry.Analysis`. There are multiple analyses, having the same recovery `recoverDeadPrimaryFunc`.

Also, the if can be simplified to `forceRefreshAllTabletsInShard(ctx, analysisEntry.AnalyzedKeyspace, analysisEntry.AnalyzedShard, checkAndRecoverFunctionCode== recoverDeadPrimaryFunc)`

Reply:
fixed